kusto/query_builder.go

```diff
@@ -506,6 +507,171 @@
+			return true
+		}
+	}
+	return false
+}
+
+// AddQuotedString escapes a string to be safely added to a stmt
+func AddQuotedString(value string, hidden bool) string {
+	if value == "" {
+		return value
+	}
+
+	var literal strings.Builder
+
+	if hidden {
+		literal.WriteString("h")
+	}
+	literal.WriteString("\"")
+	for _, c := range value {
+		switch c {
+		case '\'':
+			literal.WriteString("\\'")
+
+		case '"':
+			literal.WriteString("\\\"")
+
+		case '\\':
+			literal.WriteString("\\\\")
+
+		case '\x00':
+			literal.WriteString("\\0")
+
+		case '\a':
+			literal.WriteString("\\a")
+
+		case '\b':
+			literal.WriteString("\\b")
+
+		case '\f':
+			literal.WriteString("\\f")
+
+		case '\n':
+			literal.WriteString("\\n")
+
+		case '\r':
+			literal.WriteString("\\r")
+
+		case '\t':
+			literal.WriteString("\\t")
+
+		case '\v':
+			literal.WriteString("\\v")
+
+		default:
+			if !ShouldBeEscaped(c) {
+				literal.WriteString(string(c))
+			} else {
+				literal.WriteString(fmt.Sprintf("\\u%x", c))
+			}
+
+		}
+	}
+	literal.WriteString("\"")
+
+	return literal.String()
+}
+
+// ShouldBeEscaped Checks whether a rune should be escaped or not based on it's type.
+func ShouldBeEscaped(c int32) bool {
+	if c <= unicode.MaxLatin1 {
+		return unicode.IsControl(c)
+	}
+	return true
+}
+
+// AddInt will add an int as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddInt(query int) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%d", query)
+	return s
+}
+
+// AddFloat32 will add a Float32 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFloat32(query float32) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%f", query)
+	return s
+}
+
+// AddFloat64 will add a Float64 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFloat64(query float64) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%f", query)
+	return s
+}
+
+// AddComplex64 will add a Complex64 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddComplex64(query complex64) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%v", query)
+	return s
+}
+
+// AddComplex128 will add a Complex128 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddComplex128(query complex128) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%v", query)
+	return s
+}
+
+// AddBool will add a bool as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddBool(query bool) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%t", query)
+	return s
+}
+
+// AddByte will add a byte as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddByte(query byte) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%d", query)
+	return s
+}
+
```

Comment:
Why did we remove the AddRune method? it should just be treated as a one-length string

Reply:
as discussed with @yogilad , due to the high similarity in function to strings, we saw no real need for this method. indeed, the cost of adding it back if desired is minimal, so shouldnt be a problem if we still want to